### PR TITLE
OSD-3978 Refactor error handling & logging, reconcile logic for CCS account initialization for flexible re-queuing (pt 2)

### PR DESCRIPTION
**Pull request by clcollins**
Uses individual error variables (for clarity) and catches and returns them, to create framework for more specific error handling.  This will allow for better surfacing of errors to OCM.

Moves reconcile.Result logic for CCS into initializeNewCCSAccount, to allow flexibility in how or when the account is re-queued based on the types of errors it encounters.  This sets the stage for handling specific errors, and treating them as necessary (ie, mark account failed and do not re-queue, retry, or retry after X period of time).

Part 2 of OSD-3978 "granular status reports"

Depends on #425

### CODE REVIEW

**Comment by rogbas** [comment]
@rogbas @fahlmant please check

File: pkg/controller/account/account_controller.go [hunk 1]
```diff
@@ -156,18 +157,31 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
...
-
+	var ccsRoleID string
 	// If the account is BYOC, needs some different set up
 	if newBYOCAccount(currentAcctInstance) {
-		byocRoleID, err = r.initializeNewBYOCAccount(reqLogger, currentAcctInstance, awsSetupClient, adminAccessArn)
-		if err != nil || byocRoleID == "" {
-			r.setStateFailed(reqLogger, currentAcctInstance, err.Error())
-			reqLogger.Error(err, "Failed setting up new BYOC account")
-			return reconcile.Result{}, err
+		// TODO: Chris - check why that is
```
Comment (fahlmant) [inline pkg/controller/account/account_controller.go:166]: What is this for?

File: pkg/controller/account/account_controller.go [hunk 2]
```diff
@@ -520,6 +531,25 @@ func CreateAccount(reqLogger logr.Logger, client awsclient.Client, accountName,
 	return accountStatus, nil
 }
 
+func checkAWSAccountsLimitReached(r *ReconcileAccount, reqLogger logr.Logger, currentAccounts int) (bool, error) {
```
Comment (fahlmant) [inline pkg/controller/account/account_controller.go:534]: Is this function related to this PR?

Comment (clcollins) [reply pkg/controller/account/account_controller.go:534]: No, it's not. Not sure how that got introduced.  I'll fix it.

**Comment by clcollins** [comment]
@fahlmant Addressed your question (removed the erroneous function) and rebased.  Tested fine after changes.

**Comment by clcollins** [comment]
/retest

The test above failed on code that isn't in this PR... :thinking:
